### DISCUSSION TITLE
Is there a maximum nr of lines one can load in bCNC

### DISCUSSION THREAD

**Discussion by Herbie7** [vlachoudis/bCNC#1694] [Q&A]
I generated G code in Fusion 360 and loaded it in bCNC, which run on Lubuntu.
I noted that the outline looks good, but all slots were not displaying. When  I loaded the same file in UGS, it looked ok and ran ok.
so I wondered if there is a maximum nr of lines bCNC can manage or if there is something wrong with my installation.
Thanks

[1001.zip](https://github.com/vlachoudis/bCNC/files/8357304/1001.zip)

**Comment by Harvie** [comment]
Yes, there is timeout in seconds:

![image](https://user-images.githubusercontent.com/276504/160346088-833452c0-9786-4de2-9d86-7e039fa195c1.png)

unfortunately bCNC loads the lines very slowly, so it does have to limit the time after which it stops displaying more lines. However the file is loaded and if you run it, it will get executed all the way even when not displayed on screen.

This is very unfortunate and we need to speed things up, but python is slow. we need to rewrite the code in faster language, which is not easy.
